src/main/java/com/volmit/iris/core/TreeManager.java

```diff
@@ -0,0 +1,217 @@
+import com.volmit.iris.util.math.RNG;
+import org.bukkit.Location;
+import org.bukkit.Material;
+import org.bukkit.TreeType;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.Listener;
+import org.bukkit.event.world.StructureGrowEvent;
+import org.bukkit.util.Vector;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.Arrays;
+import java.util.Objects;
+
+public class TreeManager implements Listener {
+
+    private static final boolean debugMe = false;
+
+    public static final int maxSaplingPlane = 5;
+
+    public TreeManager() {
+        Iris.instance.registerListener(this);
+        Iris.info("Loading Sapling Manager");
+    }
+
+    /**This function does the following
+     * <br>1. Is the sapling growing in an Iris world? No -> exit</br>
+     * <br>2. Is the sapling overwriting setting on in that dimension? No -> exit</br>
+     * <br>3. Check biome for overrides for that sapling type -> Found -> use</br>
+     * <br>4. Check region ...</br>
+     * <br>5. Check dimension ...</br>
+     * <br>6. Exit if none are found</br>
+     * @param event Checks the given event for sapling overrides
+     */
+    @EventHandler
+    public void onStructureGrowEvent(StructureGrowEvent event) {
+
+        if (debugMe)
+            Iris.info("Sapling grew @ " + event.getLocation() + " for " + event.getSpecies().name() + " usedBoneMeal is " + event.isFromBonemeal());
+
+        // Must be iris world
+        if (!IrisWorlds.isIrisWorld(event.getWorld())) return;
+
+        IrisAccess worldAccess;
+        try {
+            worldAccess = Objects.requireNonNull(IrisWorlds.access(event.getWorld()));
+        } catch (Throwable e) {
+            Iris.reportError(e);
+            return;
+        }
+        IrisTreeSettings settings = worldAccess.getCompound().getRootDimension().getSaplingSettings();
+
+        if (debugMe) Iris.info("Custom saplings are enabled: " + (settings.isEnabled() ? "Yes" : "No"));
+
+        // Must have override enabled
+        if (!settings.isEnabled()) return;
+
+        KList<String> treeObjects = new KList<>();
+
+        // Get biome and region
+        IrisBiome biome = worldAccess.getBiome(event.getLocation().getBlockX(), event.getLocation().getBlockY(), event.getLocation().getBlockZ());
+        IrisRegion region = worldAccess.getCompound().getDefaultEngine().getRegion(event.getLocation().getBlockX(), event.getLocation().getBlockZ());
+
+        if (debugMe)
+            Iris.info("Biome name: " + biome.getName() + " | List of saplings: " + biome.getSaplings().toString());
+        if (debugMe)
+            Iris.info("Region name: " + region.getName() + " | List of saplings: " + region.getSaplings().toString());
+        if (debugMe)
+            Iris.info("Dimension saplings: " + settings.getSaplings().toString());
+
+        // Get sapling location
+        KList<Location> saplingLocations = getSaplingPlane(event.getLocation());
+        int saplingSize = getSaplingSize(saplingLocations);
```

Comment:
I think this could probably be optimized a bit. I think you should change this to something like this:
```java
KList<Block> locations = new KList<>();
int saplingSize = getSaplingSize(centerLoc, type, locations); 
```
The locations of all the saplings are added to the list and the center location is adjusted, all within the method
If any of the saplings are different, it will just use the previous size if it can (but not for a 1x1).

With the centerLocation known, it will be easy to translate the object

Reply:
We need to remove the saplings manually before placing the object to prevent them staying 
If an air object is placed, the saplings aren't removed, that's why I get the saplingLocations and store all of them
I could store the blocks as well I guess, but that call won't affect performance at all since it's just moving it.

Am I misunderstanding what you're saying?